platform/ios/src/MGLMapView.mm

```diff
@@ -1778,6 +1778,10 @@ - (void)handlePanGesture:(UIPanGestureRecognizer *)pan
                    default:
                       self.mbglMap.moveBy({ offset.x, offset.y }, MGLDurationFromTimeInterval(self.decelerationRate));
                 }
+            } else {
+                if ([self.delegate respondsToSelector:@selector(mapView:didCancelGestureForCamera:)]) {
+                    [self.delegate mapView:self didCancelGestureForCamera:toCamera];
+                }
```

Comment:
```suggestion
                MGLMapCamera *targetCamera = toCamera.copy;
                if ([self.delegate respondsToSelector:@selector(mapViewWillEndPanning:targetCamera:)] && [self.delegate mapViewWillEndPanning:self targetCamera:targetCamera] && ![toCamera isEqualToCamera:targetCamera]) {
                    [self setCamera:targetCamera animated:YES];
                }
```

Reply:
Actually, that would have to happen regardless of `-mapView:shouldChangeFromCamera:toCamera:`, so we’d move it out one level so it’s only conditionalized on `drift`.